pkg/pagerduty/pagerduty.go

```
@@ -98,30 +98,53 @@ func (pd *PagerDuty) formatAlert(alert *pdApi.IncidentAlert) (formatAlert Alert,
 	formatAlert.Status = alert.Status
 	formatAlert.WebURL = alert.HTMLURL
 
+	// Check if alert.Body and alert.Body["details"] exist
+	if alert.Body == nil {
+		return formatAlert, fmt.Errorf("unable to parse alert: alert body is missing")
+	}
+
+	details, ok := alert.Body["details"]
+	if !ok || details == nil {
+		return formatAlert, fmt.Errorf("unable to parse alert: alert details are missing")
+	}
+
+	detailsMap, ok := details.(map[string]interface{})
+	if !ok {
+		return formatAlert, fmt.Errorf("unable to parse alert: alert details have unexpected format")
+	}
+
 	// Check if the alert is of type 'Missing cluster'
-	isCHGM := alert.Body["details"].(map[string]interface{})["notes"]
+	isCHGM := detailsMap["notes"]
 
 	if isCHGM != nil {
-		notes := strings.Split(fmt.Sprint(alert.Body["details"].(map[string]interface{})["notes"]), "\n")
+		notes := strings.Split(fmt.Sprint(detailsMap["notes"]), "\n")
 		fmt.Print(notes)
 		formatAlert.ClusterID = strings.Replace(notes[0], "cluster_id: ", "", 1)
-		formatAlert.ClusterName = strings.Split(fmt.Sprint(alert.Body["details"].(map[string]interface{})["name"]), ".")[0]
 
-	} else {
-		formatAlert.ClusterID = fmt.Sprint(alert.Body["details"].(map[string]interface{})["cluster_id"])
-		formatAlert.ClusterName, err = pd.GetClusterName(alert.Service.ID)
+		nameValue, ok := detailsMap["name"]
+		if ok && nameValue != nil {
+			formatAlert.ClusterName = strings.Split(fmt.Sprint(nameValue), ".")[0]
+		}
 
-		// If the service mapped to the current incident is not available (404)
-		if err != nil {
-			formatAlert.ClusterName = "N/A"
+	} else {
+		clusterIDValue, ok := detailsMap["cluster_id"]
+		if ok && clusterIDValue != nil {
+			formatAlert.ClusterID = fmt.Sprint(clusterIDValue)
 		}
+
+		formatAlert.ClusterName, _ = pd.GetClusterName(alert.Service.ID)
```

Is there a reason we drop the error here? 
If an error happens, then it will be silently ignored with the cluster id  = "NA"
what if still handle err and add lodge for better visibility. 
e.g.

```// Recommended Change
formatAlert.ClusterName, err = pd.GetClusterName(alert.Service.ID)
if err != nil {
    // Replace with a proper logger warning
    fmt.Printf("WARN: Couldn't get cluster id %s: %v\n", alert.Service.ID, err)
    formatAlert.ClusterName = "N/A"
}```

---

You're right, and thanks for catching this. I initially went with the simpler approach of using a catch-all check, which works functionally. However, your point about visibility is valid - logging the error provides better operational insight.
I've updated the code to log the error at WARN level. 
I kept the catch-all check because CHGM alerts (which parse details["name"] directly) don't generate errors when the name is missing - they just leave ClusterName empty. The catch-all handles both CHGM and regular alert failures consistently.